store/storetest/link_metadata_store.go

```diff
@@ -117,9 +117,9 @@ func testLinkMetadataStoreSave(t *testing.T, ss store.Store) {
 		assert.Equal(t, result.Data.(*model.LinkMetadata).Data, &model.PostImage{Height: 10, Width: 20})
 
 		// Should return the original result, not the duplicate one
-		result = <-ss.LinkMetadata().Get(metadata.URL, metadata.Timestamp)
-		require.Nil(t, result.Err)
-		assert.Equal(t, &model.PostImage{}, result.Data.(*model.LinkMetadata).Data)
+		metadata, err := ss.LinkMetadata().Get(metadata.URL, metadata.Timestamp)
+		require.Nil(t, err)
+		assert.Equal(t, &model.PostImage{}, metadata.Data.(*model.LinkMetadata).Data)
```

Comment:
This is wrong, you are already receiving the `LinkMetadata` object, you don't need casting the object. This should be something like: `assert.Equal(t, &model.PostImage{}, metadata.Data)`